content/go/concepts/composition/composition.md

```diff
@@ -0,0 +1,141 @@
+	Toppings    []string
+	IsDelicious bool
+}
+```
+
+This struct can be used to create a new function that can help define what specific style of pizza is wanted.
+
+This is done in the following example:
+
+```go
+package main
+
+import "fmt"
+
+type Pizza struct {
+	Name        string
+	Size        string
+	Toppings    []string
+	IsDelicious bool
+}
+
+func pizzaStyle(p Pizza) string {
+	return p.Name + " pizza is a " + p.Size + " pizza with toppings of " + fmt.Sprint(p.Toppings)
+}
+
+func main() {
+	myPizza := Pizza{
+		Name:        "Margherita",
+		Size:        "medium",
+		Toppings:    []string{"tomatoes", "mozzarella", "basil"},
+		IsDelicious: true,
+	}
+
+	fmt.Println(pizzaStyle(myPizza))
+}
+```
+
+This example results in the following output:
+
+```shell
+Margherita pizza is a medium pizza with toppings of [tomatoes mozzarella basil]
+```
+
+## A Struct Within a Struct
+
+The following example has an important difference.
+
+```go
+package main
+
+import "fmt"
+
+type Pizza struct {
+	Name        string
+	Size        string
+	Toppings    []string
+	IsDelicious bool
+}
+
+func pizzaStyle(p Pizza) string {
+	return p.Name + " pizza is a " + p.Size + " pizza with toppings of " + fmt.Sprint(p.Toppings)
+}
+
+type Restaurant struct {
+	Name      string
+	Rating    int
+	PizzaMenu []Pizza
+}
+
+func restaurantInfo(r Restaurant) string {
+	return r.Name + " has a rating of " + fmt.Sprint(r.Rating) + " and serves the following pizzas: " + fmt.Sprint(r.PizzaMenu)
+}
+
+func main() {
+	myPizza := Pizza{
+		Name:        "Margherita",
+		Size:        "medium",
+		Toppings:    []string{"tomatoes", "mozzarella", "basil"},
+		IsDelicious: true,
+	}
+
+	myRestaurant := Restaurant{
+		Name:      "Pizzeria del Corso",
+		Rating:    4,
+		PizzaMenu: []Pizza{myPizza},
+	}
+
+	fmt.Println(pizzaStyle(myPizza))
+	fmt.Println(restaurantInfo(myRestaurant))
+}
+```
+
+In this example the `Restaurant` struct contains the `Pizza` struct.
+
+In this example, a struct called `Pizza` is defined with fields for the name, size, toppings, and whether or not it's delicious. The `pizzaStyle` function takes a `Pizza` struct as an argument and returns a string that describes the pizza.
+
+A struct called `Restaurant` is also defined that contains the `Pizza` struct, the restaurant name, and the restaurant rating. The `restaurantInfo` function takes a `Restaurant` struct as an argument and returns a string that describes the restaurant.
+
+In the main function, a `Pizza` struct called `myPizza` and a `Restaurant` struct called `myRestaurant` are created. Then `myPizza` is passsed to `pizzaStyle` and `myRestaurant` is passed to `restaurantInfo`, which are then printed to the console.
+
+This example results in the following output:
+
+```shell
+Margherita pizza is a medium pizza with toppings of [tomatoes mozzarella basil]
+Pizzeria del Corso has a rating of 4 and serves the following pizzas: [{Margherita medium [tomatoes mozzarella basil] true}]
+```
+
+## Conclusions
```

Comment:
Can we change this heading to something more meaningful to go along with what's discussed in this section? Perhaps benefits of using composition or something along those lines.